### ISSUE TITLE
idea for cleaner database structure

### ISSUE THREAD

**Issue by michielbdejong** [tosdr/phoenix#185]
* each point has one service, that makes some things a lot simpler
* each point has one case, which determines how the point affects the rating of the service it applies to
* a point can have multiple topics, they just act as tags
* each point, case, service, and topic has a discussion, which can be partially the imported discussion from Google groups, and partially new discussion to be added via phoenix
* each point, case, service, and topic has a change log where you can see how it was updated over time, and by whom

@piks3l let's have a discussion about this? I'll ping you on irc.

**Comment by piks3l** [comment]
I need to review the current db model, I'll tell you later today. 

For the early dev of phoenix, we slashed through the previous db model to make a more simple one, see (the wiki)[https://github.com/tosdr/phoenix/wiki/Database]. I saw the previous system as too complex although maybe more complete. 

Nonetheless, changing the model is easy, we just need to run migration. But I'd like to discuss on some of those things. When do you think we could do that? Thursday morning?

**Comment by michielbdejong** [comment]
Sure! What time?

**Comment by piks3l** [comment]
9am Brussels time ?

**Comment by piks3l** [comment]
Running a bit late, give me 15min

**Comment by michielbdejong** [comment]
notes from that phone call: https://matrix.org/_matrix/media/v1/download/matrix.org/YRqkvzkRWvscqIkvkUsgBQbl

**Comment by michielbdejong** [comment]
Idea for database schema, that makes it possible to keep the [current api](https://github.com/tosdr/tosdr.org/wiki/Specification:-points):

### Point
* id, title: same as in old db
* discussion: import into newly created point_comments table
* "services" array: reduce to one "service", split up points that have multiple
* topics: change from string to foreign key on topics table. These will just act as labels
* tosdr
  * point ("classification"), score, tldr: move to newly created 'Case' table
  * case: change from string to foreign key on cases table
  * irrelevant, disputed, reason: move to point_comments table. when a point is irrelevant or disputed, it has no case. the difference can be seen in the point_comments table.
* sources: not implemented yet (we would start using links to tosback here)
* binding: move to cases table
* meta:
  * spec-version, license-for-this-file: create on the fly in api view
  * author, contributors: move to individual comments in point_comments table

### Point_comments
* author: foreign key on User
* point: foreign key on Point
* date: when the comment was made
* action: can be assigning a case, removing a case, merging, splitting, etc, or just commenting.
* text, etc.: for now, no further machine-readable columns here, just stuff to display

### Case
* classification (aka "point": blocker / bad / neutral / good )
* score (0..100 or change to 0..10 if we divide everything by 10?)
* tldr (needs to be written without mentioning the name of the service)

### Service
* id, name: same as in old db
* type: i think we can delete type 'software' and just focus on type 'service'
* urls: link to tosback
* maybe add logos or logo urls to db?
* tosdr.rated would be a cached value from the output of https://github.com/tosdr/tosdr-build/blob/master/scripts/checkclasses.js
* rest more less like https://github.com/tosdr/tosdr.org/wiki/Specification:-services

### Topics
* more or less like https://github.com/tosdr/tosdr.org/wiki/Specification:-topics

**Comment by michielbdejong** [comment]
I'll write a migration for this

**Comment by michielbdejong** [comment]
rails generate model Case classification:string score:integer tldr:string
rails generate model Comment point:references summary:string
rails generate migration AddRefsToPoints case:references service:references topics:references

**Comment by michielbdejong** [comment]
Continued in https://github.com/tosdr/phoenix/pull/189
